OpenRA.Mods.Common/Widgets/Logic/Editor/ActorEditLogic.cs

```diff
@@ -125,12 +125,15 @@ public ActorEditLogic(Widget widget, World world, WorldRenderer worldRenderer, D
 					return;
 				}
 
-				// Check for duplicate actor ID
-				if (CurrentActor.ID.Equals(actorId, StringComparison.OrdinalIgnoreCase))
+				// Check for duplicate ActorID
+				foreach (var kv in world.Map.ActorDefinitions)
 				{
-					if (editorActorLayer[actorId] != null)
```

Comment:
Is there a difference between the actors in `world.Map.ActorDefinitions` and `editorActorLayer`? I'd expect them to be the same initially, but only `editorActorLayer` getting updated while working on the map.

Reply:
Thinking about it, the real error here could only be the missing `!` in front of the first if, or am I missing something else?

Reply:
I'm very new to all of this, so please bear with me.  What you have said matches my observations.  I don't think I understand your question/concern?

Reply:
We definitely shouldn't be using `world.Map.ActorDefinitions` here. You are correct that this is only used during the initial map load and (re)saving.

Reply:
@chrisharris333 my apologies. To be more clear: I was wondering if just changing `CurrentActor.ID.Equals(actorId, StringComparison.OrdinalIgnoreCase)` to `!CurrentActor.ID.Equals(actorId, StringComparison.OrdinalIgnoreCase)` and keeping the `editorActorLayer[actorId] != null` check is enough to fix the issue.

Reply:
@abcdefg30 I'll test that out.

Reply:
Doesn't work just adding the !.  @pchote Is there a list of actors that does get updated as the map gets worked on?

Reply:
The `if (editorActorLayer[actorId] != null)` is already doing the right thing here.

Reply:
Ok, I figured out why it didn't appear to be doing anything.  Stay Tuned.

Reply:
So it was working as expected, but it just wasn't doing anything.  The few new lines I added do make the "Duplicate ActorID" message actually show up now, with the original detection code.

Reply:
@abcdefg30 and @pchote please take another look when you get the chance.  Thanks.